### PR TITLE
Fix Crash in `BottomSheetViewController`

### LINKED ISSUE #10250: [Crash] Crash in the bottom sheet for adding more details to a product
**Describe the bug**
I encountered a crash on iOS 16.4, simulator iPhone 14 Pro Max, while I was trying to add a new product. 

**Steps**
1) Tap add new physical product
2) Tap "Add more details"
3) Tap the "top line" in the bottom sheet.
4) Crash

I was able to reproduce this constantly.

**Screen**

**Logs**
```
2023-07-17 19:34:35.456149+0200 WooCommerce[27121:600935] [TableView] Changing the background color of UITableViewHeaderFooterView is not supported. Use the background view configuration instead.
2023-07-17 19:37:11.400739+0200 WooCommerce[27121:609733] Received XPC error Connection interrupted for message type 3 kCFNetworkAgentXPCMessageTypePACQuery
2023-07-17 19:37:11.401764+0200 WooCommerce[27121:609733] Received XPC error Connection invalid for message type 3 kCFNetworkAgentXPCMessageTypePACQuery
2023-07-17 19:39:03.271770+0200 WooCommerce[27121:614478] 🔵 Tracked add_product_product_type_selected, properties: [AnyHashable("creation_type"): "manual", AnyHashable("is_virtual"): false, AnyHashable("product_type"): "simple", AnyHashable("blog_id"): 171745248, AnyHashable("is_wpcom_store"): false]
2023-07-17 19:39:03.574062+0200 WooCommerce[27121:612995] 🔵 Tracked product_detail_loaded, properties: [AnyHashable("blog_id"): 171745248, AnyHashable("has_linked_products"): false, AnyHashable("is_wpcom_store"): false]
2023-07-17 19:39:38.738185+0200 WooCommerce[27121:600935] [TableView] Changing the background color of UITableViewHeaderFooterView is not supported. Use the background view configuration instead.
2023-07-17 19:39:39.578092+0200 WooCommerce[27121:600935] +[WordPressUI.BottomSheetViewController buttonPressed]: unrecognized selector sent to class 0x10ebc50b0
2023-07-17 19:39:39.603089+0200 WooCommerce[27121:600935] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '+[WordPressUI.BottomSheetViewController buttonPressed]: unrecognized selector sent to class 0x10ebc50b0'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000115ae6330 __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x000000010de3d274 objc_exception_throw + 56
	2   CoreFoundation                      0x0000000115af4d3c __CFExceptionProem + 0
	3   CoreFoundation                      0x0000000115aea224 ___forwarding___ + 1308
	4   CoreFoundation                      0x0000000115aec50c _CF_forwarding_prep_0 + 92
	5   UIKitCore                           0x000000012f1d426c -[UIApplication sendAction:to:from:forEvent:] + 96
	6   Sentry                              0x000000010df43014 __49-[SentrySwizzleWrapper swizzleSendAction:forKey:]_block_invoke_2 + 200
	7   UIKitCore                           0x000000012eb7e590 -[UIControl sendAction:to:forEvent:] + 108
	8   UIKitCore                           0x000000012eb7e8d4 -[UIControl _sendActionsForEvents:withEvent:] + 268
	9   UIKitCore                           0x000000012eb7b578 -[UIButton _sendActionsForEvents:withEvent:] + 120
	10  UIKitCore                           0x000000012eb7d604 -[UIControl touchesEnded:withEvent:] + 392
	11  UIKitCore                           0x000000012f209638 -[UIWindow _sendTouchesForEvent:] + 900
	12  UIKitCore                           0x000000012f20aadc -[UIWindow sendEvent:] + 3180
	13  UIKitCore                           0x000000012f1e8ae4 -[UIApplication sendEvent:] + 692
	14  UIKitCore                           0x000000012f264644 __dispatchPreprocessedEventFromEventQueue + 1880
	15  UIKitCore                           0x000000012f267590 __processEventQueue + 5556
	16  UIKitCore                           0x000000012f2604b8 __eventFetcherSourceCallback + 156
	17  CoreFoundation                      0x0000000115a49c6c __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
	18  CoreFoundation                      0x0000000115a49bb4 __CFRunLoopDoSource0 + 172
	19  CoreFoundation                      0x0000000115a49324 __CFRunLoopDoSources0 + 232
	20  CoreFoundation                      0x0000000115a43958 __CFRunLoopRun + 748
	21  CoreFoundation                      0x0000000115a43254 CFRunLoopRunSpecific + 584
	22  GraphicsServices                    0x000000011c667c9c GSEventRunModal + 160
	23  UIKitCore                           0x000000012f1ceff0 -[UIApplication _run] + 868
	24  UIKitCore                           0x000000012f1d2f3c UIApplicationMain + 124
	25  WooCommerce                         0x0000000104c0b898 main + 496
	26  dyld                                0x0000000109945514 start_sim + 20
	27  ???                                 0x0000000109b71f28 0x0 + 4457963304
	28  ???                                 0x416f000000000000 0x0 + 4714987334880198656
)
libc++abi: terminating due to uncaught exception of type NSException
(lldb) 
```

### PR DESCRIPTION
**Pull request by pmusolino**
Fixes: #10250 

## Description
I encountered a crash on iOS 16.4, simulator iPhone 14 Pro Max, while I was trying to add a new product. In the end, I discovered that this crash happens in every screen where we use the `BottomSheetViewController` component from the WordPressUI library.

## Testing instructions
1. Tap add new physical product
2. Tap "Add more details"
3. Tap the "top line" in the bottom sheet.
4. Crash

Run this PR, you should not be able to expact a crash anymore. 

## Before merging to trunk
- [x] Point WordPressUI pod to the release 1.13.1

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by itsmeichigo — APPROVED** [review]
:shipit:

File: Podfile [hunk 1]
```diff
@@ -90,7 +90,7 @@ target 'WooCommerce' do
 
   wordpress_shared
 
-  pod 'WordPressUI', '~> 1.13'
+  pod 'WordPressUI', :git => 'https://github.com/wordpress-mobile/WordPressUI-iOS.git', :branch => 'task/fix-crash-BottomSheetViewController'
```
Comment (itsmeichigo) [inline Podfile]: Please point this to the library's `trunk` after merging the fix.

Comment (mokagio) [reply Podfile]: @itsmeichigo @pmusolino 1.13.1 has been shipped and this branch now points to it.

Comment (pmusolino) [reply Podfile]: Thank you both!

**Pull request merged**